public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -467,7 +467,7 @@ export function processRangeQueryResponse(
   switch (response.data.resultType) {
     case LokiResultType.Stream:
       return of({
-        data: lokiStreamsToDataframes(response.data.result, target, limit, config, reverse),
+        data: lokiStreamsToDataframes(response.data.result.slice(0, limit), target, limit, config, reverse),
```

Comment:
Do we need to slice here? Loki should be limiting the returns itself right? Or is there a case it will return more data than the limit?

Reply:
Yep, we need to slice the result as Loki doesn't limit results properly now (https://github.com/grafana/loki/issues/1604) on negative limit values. Opened an issue, when we get it fixed we can remove the slice there.

Reply:
@Estrax and why do we need to send a negative number as limit to loki? We should be doing validation for that. I though we treat negative values as 0, don't we?

Reply:
Yes, fixed that on our end.